pkg/sql/tests/add_column_generated_test.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	gosql "database/sql"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/stretchr/testify/require"
+)
+
+// TestAddGeneratedColumnValidation tests if validation
+// when creating computed columns is working properly
+func TestAddGeneratedColumnValidation(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+
+	var s serverutils.TestServerInterface
+	params, _ := CreateTestServerParams()
+
+	var db *gosql.DB
+	s, db, _ = serverutils.StartServer(t, params)
+	//sqlDB := sqlutils.MakeSQLRunner(db)
+	defer s.Stopper().Stop(ctx)
+
+	_, err := db.Exec(`
+	CREATE TABLE t (i INT PRIMARY KEY);
+	INSERT INTO t VALUES (1);
+	CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi');`)
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = db.Exec(`ALTER TABLE t ADD COLUMN s VARCHAR(2) AS ('st') STORED;`)
```

Comment:
We can write all these tests as logic tests, which would be preferred. Can you convert all these cases to logic tests in either `pkg/sql/logictest/testdata/logic_test/computed` or `pkg/sql/logictest/testdata/logic_test/virtual`?

Also, we need cases covering CREATE TABLE as well.